### PR TITLE
e2e: add serial lane skeleton and infrastructure

### PR DESCRIPTION
**Pull request by fromanirh**
Spin off from #95: bootstrap the serial lane and add all the utilities which we know we need, because we will use them in #95, but are likely helpful in other related PRs like #100

### CODE REVIEW

File: test/utils/objects/wait/deployment.go
Comment (jlojosnegros) [inline test/utils/objects/wait/deployment.go:47]: None of these functions are waiting, so maybe it could be a little bit confusing when call them using wait package.

Comment (fromanirh) [reply test/utils/objects/wait/deployment.go:47]: may be worth moving them back in the parent package, but this will cause (or make much more likely) circular deps. Let's see.

**Comment by fromanirh** [comment]
all the code here is actually being used and exercised in #95 but since #95 needs to be on hold for a little while, I suggest to unblock other work by merging this PR which will provide quite a lot of shared infrastructure

**Review by jlojosnegros — APPROVED** [review]
/lgtm

File: test/utils/noderesourcetopologies/noderesourcetopologies.go
Comment (Tal-or) [inline test/utils/noderesourcetopologies/noderesourcetopologies.go:129]: Do we care only about limits?
What if the pod is burstable and only have requests

File: test/utils/noderesourcetopologies/noderesourcetopologies.go
Comment (Tal-or) [inline test/utils/noderesourcetopologies/noderesourcetopologies.go:180]: The name of the package is already ```noderesourcetopologies``` so we can remove the name itself from the function.
Then it is going to be called like ```noderesourcetopologies.FindFromList```

File: test/utils/noderesourcetopologies/noderesourcetopologies.go
Comment (Tal-or) [inline test/utils/noderesourcetopologies/noderesourcetopologies.go:159]: dito

File: test/utils/noderesourcetopologies/noderesourcetopologies.go
Comment (Tal-or) [inline test/utils/noderesourcetopologies/noderesourcetopologies.go:34]: The name of the package is already ```noderesourcetopologies``` so we can remove the name from the function.
Then it is going to be called like ```noderesourcetopologies.GetUpdated```

File: test/utils/fixture/fixture.go
Comment (fromanirh) [reply test/utils/fixture/fixture.go:50]: we can use `Errorf` and return `(*Fixture, error)`, yes

File: test/utils/nrosched/nrosched.go
Comment (fromanirh) [reply test/utils/nrosched/nrosched.go:56]: sure thing

File: test/utils/noderesourcetopologies/noderesourcetopologies.go
Comment (fromanirh) [reply test/utils/noderesourcetopologies/noderesourcetopologies.go:129]: for the current use cases yes :) I'll rename the function to make evident we deal only with the Guaranteed pods atm.

File: test/utils/noderesourcetopologies/noderesourcetopologies.go
Comment (fromanirh) [reply test/utils/noderesourcetopologies/noderesourcetopologies.go:180]: ack

File: test/utils/noderesourcetopologies/noderesourcetopologies.go
Comment (fromanirh) [reply test/utils/noderesourcetopologies/noderesourcetopologies.go:198]: true, even though this is an internal function, so it will be called without the prefix. I think `zoneResourcesMatchesRequest` is good enough anyway.

File: test/utils/noderesourcetopologies/noderesourcetopologies.go
Comment (fromanirh) [reply test/utils/noderesourcetopologies/noderesourcetopologies.go:146]: ack

File: test/utils/noderesourcetopologies/noderesourcetopologies.go
Comment (fromanirh) [reply test/utils/noderesourcetopologies/noderesourcetopologies.go:159]: ack

File: test/utils/noderesourcetopologies/noderesourcetopologies.go
Comment (fromanirh) [reply test/utils/noderesourcetopologies/noderesourcetopologies.go:34]: ack

**Comment by fromanirh** [comment]
/retest
I still think it's more likely a cluster issue, as none of this code is likely to trigger this failure

**Comment by fromanirh** [comment]
ok so I'm actually introducing a bug. Will check.

**Comment by fromanirh** [comment]
/hold cancel
this is the safe set, if CI passes we're good to go

**Pull request merged**
